services/server/src/server/services/StorageService.ts

```diff
@@ -0,0 +1,67 @@
+import { Match, CheckedContract } from "@ethereum-sourcify/lib-sourcify";
+import {
+  IpfsRepositoryService,
+  IpfsRepositoryServiceOptions,
+} from "./storageServices/IpfsRepositoryService";
+import {
+  AllianceDatabaseService,
+  AllianceDatabaseServiceOptions,
+} from "./storageServices/AllianceDatabaseService";
+import { logger } from "../../common/logger";
+
+export interface IStorageService {
+  init(): Promise<boolean>;
+  storeMatch(contract: CheckedContract, match: Match): Promise<void | Match>;
+  checkByChainAndAddress?(address: string, chainId: string): Match[];
+  checkAllByChainAndAddress?(address: string, chainId: string): Match[];
+}
+
+interface StorageServiceOptions {
+  ipfsRepositoryServiceOptions: IpfsRepositoryServiceOptions;
+  allianceDatabaseServiceOptions?: AllianceDatabaseServiceOptions;
+}
+
+export class StorageService {
+  ipfsRepository: IpfsRepositoryService;
+  allianceDatabase?: AllianceDatabaseService;
+
+  constructor(options: StorageServiceOptions) {
+    this.ipfsRepository = new IpfsRepositoryService(
+      options.ipfsRepositoryServiceOptions
+    );
+    if (
+      options.allianceDatabaseServiceOptions?.googleCloudSql ||
+      options.allianceDatabaseServiceOptions?.postgres
+    ) {
+      this.allianceDatabase = new AllianceDatabaseService(
+        options.allianceDatabaseServiceOptions
+      );
+    }
+  }
+
+  async init() {
+    try {
+      await this.ipfsRepository?.init();
+    } catch (e: any) {
+      throw new Error("Cannot initialize ipfsRepository: " + e.message);
+    }
+    try {
+      await this.allianceDatabase?.init();
```

Comment:
Should we execute init? I won't init no? @marcocastignoli

Reply:
I would comment the `allianceDatabase.init()` call in any case, just to make it explicit that it's disabled